webapp/src/js/controllers/reports.js

```diff
@@ -49,6 +51,34 @@ angular
 
     var lineage = lineageFactory();
 
+    // Render the facilities hierarchy as the user is scrolling through the list
+    // Initially, don't load/render any
+    $scope.totalFacilitiesDisplayed = 0;
+    $scope.facilities = [];
+
+    // Load the facilities hierarchy and render one district hospital 
+    // when the user clicks on the filter dropdown
+    $scope.monitorFacilityDropdown = () => {
+      PlaceHierarchy()
+        .then(function(hierarchy) {
+          $scope.facilities = hierarchy;
+          $scope.totalFacilitiesDisplayed += 1;
+        })
+        .catch(function(err) {
+          $log.error('Error loading facilities', err);
+        });
+
+      $document[0].querySelector('#facilityDropdown span.dropdown-menu > ul')
```

Comment:
I think it would be cleaner to use the jquery `$` syntax as this is what you're doing under the hood anyway. eg:

```js
$('#facilityDropdown span.dropdown-menu > ul')...
```